examples/summarization/scripts/reward_summarization.py

```diff
@@ -0,0 +1,198 @@
+        },
+    )
+    per_device_train_batch_size: Optional[int] = field(default=16)
+    per_device_eval_batch_size: Optional[int] = field(default=16)
+    gradient_accumulation_steps: Optional[int] = field(default=4)
+    learning_rate: Optional[int] = field(default=2e-5)
+    model_name: Optional[str] = field(
+        default="gpt2",
+        metadata={
+            "help": "The model that you want to train from the Hugging Face hub. E.g. gpt2, gpt2-xl, bert, etc."
+        },
+    )
+    bf16: Optional[bool] = field(
+        default=False,
+        metadata={
+            "help": "This essentially cuts the training time in half if you want to sacrifice a little precision and have a supported GPU."
+        },
+    )
+
+
+parser = HfArgumentParser(ScriptArguments)
+script_args = parser.parse_args_into_dataclasses()[0]
+
+# Load the human comparisons dataset for tuning the reward model.
+ds = load_dataset("openai/summarize_from_feedback", name="comparisons")
+
+# Define the training args. Needs to be done before the model is loaded if you are using deepspeed.
+training_args = TrainingArguments(
+    output_dir=f"{script_args.model_name}_summarization_reward_model",
+    learning_rate=script_args.learning_rate,
+    per_device_train_batch_size=script_args.per_device_train_batch_size,
+    per_device_eval_batch_size=script_args.per_device_eval_batch_size,
+    num_train_epochs=5,
+    weight_decay=0.001,
+    evaluation_strategy="epoch",
+    save_strategy="epoch",
+    gradient_accumulation_steps=script_args.gradient_accumulation_steps,
+    load_best_model_at_end=True,
+    push_to_hub=True,
+    deepspeed=script_args.deepspeed,
+    local_rank=script_args.local_rank,
+    remove_unused_columns=False,
+    label_names=[],
+)
+
+# Load the value-head model and tokenizer.
+tokenizer = AutoTokenizer.from_pretrained(script_args.model_name)
+model = AutoModelForSequenceClassification.from_pretrained(script_args.model_name, num_labels=1)
+
+# Need to do this for gpt2, because it doesn't have an official pad token.
+tokenizer.pad_token = tokenizer.eos_token
+model.config.pad_token_id = tokenizer.eos_token_id
+
+# Turn the dataset into pairs of post + summaries, where text_j is the preferred post + summary and text_k is the other.
+def turn_into_text_classification_format(examples):
+    new_examples = {"text_j": [], "text_k": []}
+    for info, summaries, choice in zip(examples["info"], examples["summaries"], examples["choice"]):
+        assert len(summaries) == 2
+        assert choice in (0, 1)
+        original_text_field = "post" if info["post"] is not None else "article"
+        new_examples["text_j"].append(
+            summaries[choice]["text"] + " " + tokenizer.bos_token + " " + info[original_text_field]
+        )
+        new_examples["text_k"].append(
+            summaries[0 if choice == 1 else 1]["text"] + " " + tokenizer.bos_token + " " + info[original_text_field]
+        )
+
+    return new_examples
+
+
+num_proc = (
+    100
```

Comment:
it looks like a very quick operation so going for 100 processes might actually be more overhead than if we went maybe for 8?

Reply:
yeah good point. The dataset size is fixed, after all.